app/models/concerns/patient_reportable.rb

```diff
@@ -1,18 +1,35 @@
 module PatientReportable
   extend ActiveSupport::Concern
+  LTFU_PERIOD = 12.months
 
   included do
     scope :with_diabetes, -> { joins(:medical_history).merge(MedicalHistory.diabetes_yes).distinct }
     scope :with_hypertension, -> { joins(:medical_history).merge(MedicalHistory.hypertension_yes).distinct }
 
     scope :excluding_dead, -> { where.not(status: :dead) }
+    scope :excluding_ltfu, ->(ltfu_as_of: Date.today) do
+      where(id: latest_bp_within_ltfu_period(ltfu_as_of).select(:patient_id))
+    end
 
-    scope :for_reports, ->(with_exclusions: false) do
+    scope :for_reports, ->(with_exclusions: false, exclude_ltfu_as_of: nil) do
       if with_exclusions
-        with_hypertension.excluding_dead
+        if exclude_ltfu_as_of
```

Comment:
I believe we could avoid this branching logic by rewriting this as:
`
skope = with_hypertension
skope = skope.excluding_dead if with_exclusions
skope = skope.excluding_ltfu(ltfu_as_of: exclude_ltfu_as_of) if exclude_ltfu_as_of
skope
`
Admittedly, this could cause issues if `with_exclusions` is false and `exclude_ltfu_as_of` is truthy, but that shouldn't happen in principle and could be prevented.

Reply:
We're planning on enabling exclusions all across at once with a flipper flag, so there _will_ be an interim period when `exclude_ltfu_as_of` will be set and `with_exclusions` will be false. It doesn't look pretty but  `with_exclusions` will get cleaned up after release.